src/main/java/Duke.java

```diff
@@ -1,10 +1,233 @@
+            }
+        }
+        System.out.println(LINE_DIVIDER);
+    }
+
+    private static void addTodo(ArrayList<Task> tasks, String description) {
+        Todo todo = new Todo(description);
+        tasks.add(todo);
+        printTaskAdded(tasks, todo);
+    }
+
+    private static void addDeadline(ArrayList<Task> tasks, String description, String dueDate) {
+        Deadline deadline = new Deadline(description, dueDate);
+        tasks.add(deadline);
+        printTaskAdded(tasks, deadline);
+    }
+
+    private static void addEvent(ArrayList<Task> tasks, String description, String eventDate) {
+        Event event = new Event(description, eventDate);
+        tasks.add(event);
+        printTaskAdded(tasks, event);
+    }
+
+    private static void markTaskAsDone(ArrayList<Task> tasks, int taskNumber) {
+        if (taskNumber < 1 || taskNumber > tasks.size()) {
+            printInvalidArgumentsMessage();
+            return;
+        }
+        /* change from 1-based indexing to 0-based indexing */
+        taskNumber = taskNumber - 1;
+        tasks.get(taskNumber).markAsDone();
+        System.out.println(LINE_DIVIDER);
+        System.out.println("\t Nice! I've marked this task as done: ");
+        System.out.println("\t   " + tasks.get(taskNumber));
+        System.out.println(LINE_DIVIDER);
+    }
+
+    private static void printTaskAdded(ArrayList<Task> tasks, Task task) {
+        System.out.println(LINE_DIVIDER);
+        System.out.println("\t added: " + task);
+        System.out.println(getNumTasksString(tasks.size()));
+        System.out.println(LINE_DIVIDER);
+    }
+
+    private static void printFallbackMessage() {
+        System.out.println(LINE_DIVIDER);
+        System.out.println("\t I didn't quite catch what you were saying. Please try again.");
+        System.out.println("\t Try using \"help\" for a list of commands.");
+        System.out.println(LINE_DIVIDER);
+    }
+
+    private static void printEmptyInputMessage() {
+        System.out.println(LINE_DIVIDER);
+        System.out.println("\t Please enter a command!");
+        System.out.println(LINE_DIVIDER);
+    }
+
+    private static void printInsufficientArgumentsMessage() {
+        System.out.println(LINE_DIVIDER);
+        System.out.println("\t Please give me more details about the task!");
+        System.out.println(LINE_DIVIDER);
+    }
+
+    private static void printInvalidArgumentsMessage() {
+        System.out.println(LINE_DIVIDER);
+        System.out.println("\t That's an invalid task number!");
+        System.out.println(LINE_DIVIDER);
+    }
+
+    private static void interactWithUser() {
+        Scanner userInput = new Scanner(System.in);
+        ArrayList<Task> tasks = new ArrayList<Task>();
+        boolean isDoneReadingInputs = false;
+        while (!isDoneReadingInputs) {
+            String userCommand = userInput.nextLine();
+            String[] commandTokens = tokenizeInput(userCommand);
```

Comment:
Hi, I like the way you tokenize your inputs before passing them into the function. And maybe assigning them to a specific string is before passing would be easier to read compared to the array item itself?